### PR TITLE
Support passwordless ssh for pdu

### PR DESCRIPTION
**Pull request by cxhong**
for Epic #3516 ,   use rspconfig command to config ssh passwordless for PDU
usage:

```
#rspconfig c685f2coralpdu sshcfg
# ssh c685f2coralpdu
Warning: Permanently added 'c685f2coralpdu' (ECDSA) to the list of known hosts.
Last login: Sun Jan 15 00:16:43 2017 from 172.10.1.5
root@c685f2coralpdu:~#

```

### CODE REVIEW

File: xCAT-server/lib/xcat/plugins/pdu.pm [hunk 1]
```diff
@@ -457,6 +467,123 @@ sub connectTopdu {
 
 }
 
+sub process_sshcfg { 
+    my $noderange = shift;
+    my $subcmd = shift;
+    my $callback = shift;
+
+    my $password = "password8";
```
Comment (zet809) [inline xCAT-server/lib/xcat/plugins/pdu.pm]: The default password for PDU?
Do we have any table to store PDU account other than root?

File: xCAT-server/lib/xcat/plugins/pdu.pm [hunk 2]
```diff
@@ -457,6 +467,123 @@ sub connectTopdu {
...
+
+    my $nodetab = xCAT::Table->new('hosts');
+    my $nodehash = $nodetab->getNodesAttribs($noderange,['ip','otherinterfaces']);
+    
+    foreach my $pdu (@$noderange) {
+        my $msg = " process_sshcfg";
+        xCAT::SvrUtils::sendmsg($msg, $callback, $pdu, %allerrornodes);
+
+        #remove old host key from /root/.ssh/known_hosts
+        $cmd = `ssh-keygen -R $pdu`;
```
Comment (zet809) [inline xCAT-server/lib/xcat/plugins/pdu.pm]: Use xCAT::Utils->runcmd to run the command will be better.

File: xCAT-server/lib/xcat/plugins/pdu.pm [hunk 3]
```diff
@@ -457,6 +467,123 @@ sub connectTopdu {
...
+            xCAT::SvrUtils::sendmsg($msg, $callback, $pdu, %allerrornodes);
+            next;
+        }
+
+        my $ret;
+        my $err;
+
+        ($ret, $err) = session_exec($exp, "mkdir -p /home/root/.ssh");
+        ($ret, $err) = session_exec($exp, "chmod 700 /home/root/.ssh");
+        ($ret, $err) = session_exec($exp, "echo \"$rootkey\" >/home/root/.ssh/authorized_keys");
```
Comment (zet809) [inline xCAT-server/lib/xcat/plugins/pdu.pm]: The file authorized_keys will be rewrite whenever configuration keys, the keys for other MN or SN will be removed. Is this what we want?

File: xCAT-server/lib/xcat/plugins/pdu.pm (hunk 1 above)
Comment (cxhong) [inline xCAT-server/lib/xcat/plugins/pdu.pm]: yeh, this is default PDU password.  We don't have other account other than root.  I can move this to password table.

File: xCAT-server/lib/xcat/plugins/pdu.pm (hunk 2 above)
Comment (cxhong) [inline xCAT-server/lib/xcat/plugins/pdu.pm]: modified

File: xCAT-server/lib/xcat/plugins/pdu.pm (hunk 3 above)
Comment (cxhong) [inline xCAT-server/lib/xcat/plugins/pdu.pm]: I think that should be fine.  we only need xdsh command to work.

File: xCAT-server/lib/xcat/plugins/pdu.pm (hunk 1 above)
Comment (cxhong) [inline xCAT-server/lib/xcat/plugins/pdu.pm]: I opened issue #3534 for this.  need to modify the pdu table.

File: xCAT-server/lib/xcat/plugins/pdu.pm [hunk 4]
```diff
@@ -487,7 +487,8 @@ sub process_sshcfg {
         xCAT::SvrUtils::sendmsg($msg, $callback, $pdu, %allerrornodes);
 
         #remove old host key from /root/.ssh/known_hosts
-        $cmd = `ssh-keygen -R $pdu`;
+        $cmd = "ssh-keygen -R $pdu";
+        my $result = xCAT::Utils->runcmd($cmd, 0);
```
Comment (zet809) [inline xCAT-server/lib/xcat/plugins/pdu.pm]: Hi, @cxhong , if we don't care about the result, why do we need to get the $result?

File: xCAT-server/lib/xcat/plugins/pdu.pm (hunk 3 above)
Comment (zet809) [inline xCAT-server/lib/xcat/plugins/pdu.pm]: OK, thx!

File: xCAT-server/lib/xcat/plugins/pdu.pm (hunk 1 above)
Comment (zet809) [inline xCAT-server/lib/xcat/plugins/pdu.pm]: +1

File: xCAT-server/lib/xcat/plugins/pdu.pm (hunk 4 above)
Comment (cxhong) [inline xCAT-server/lib/xcat/plugins/pdu.pm]: modified

**Comment by zet809** [comment]
Looks good to me, only concern about the performance. Since it is done for a group of nodes one by one.

**Pull request merged**